tests/rpc-tests/rpc-test.js

```diff
@@ -0,0 +1,777 @@
+          response = await axiosInstance.post('/', getTestData[i].req)
+          assert.deepStrictEqual(
+            currentSnapshotResponse.data.result.validatorSet.validators,
+            response.data.result
+          )
+          continue
+        }
+
+        if (getTestData[i].req.method === 'eth_createAccessList') {
+          assert.deepStrictEqual(
+            sortAccessList(response.data.result.accessList),
+            sortAccessList(getTestData[i].res.result.accessList)
+          )
+          assert.deepStrictEqual(
+            response.data.result.gasUsed,
+            getTestData[i].res.result.gasUsed
+          )
+        } else {
+          assert.deepStrictEqual(response.data, getTestData[i].res)
+        }
+      }
+    }
+
+    // Sender rpc calls
+    const sendTestData = fetchTestFiles(
+      `../../tests/rpc-tests/RPC-testdata/${process.env.RPC_NETWORK}/senders`
+    )
+
+    let nonce = await web3.eth.getTransactionCount(sender)
+
+    const getFilterChanges = {
+      method: 'eth_getFilterChanges',
+      params: [''],
+      id: 1,
+      jsonrpc: '2.0'
+    }
+    let filterResponse
+
+    const getFilterLogs = {
+      method: 'eth_getFilterLogs',
+      params: [''],
+      id: 1,
+      jsonrpc: '2.0'
+    }
+    let filterLogsResponse
+
+    let senderrIterattions = process.env.EXECUTION_COUNT_SENDERS
+    if (
+      senderrIterattions <= 0 ||
+      senderrIterattions === null ||
+      senderrIterattions === undefined
+    ) {
+      senderrIterattions = 1
+    }
+
+    // Finally, fire 'em
+    for (let iter = 0; iter < senderrIterattions; iter++) {
+      await updateSenderTestData(
+        sendTestData,
+        web3,
+        sender,
+        nonce,
+        axiosInstance
+      )
+      console.log('📍Executing sender rpc calls. Iteration: ', iter)
+      for (let i = 0; i < sendTestData.length; i++) {
+        if (
+          process.env.RPC_URL &&
+          (sendTestData[i].req.method === 'eth_sendRawTransaction' ||
+            sendTestData[i].req.method === 'eth_sign' ||
+            sendTestData[i].req.method === 'eth_signTransaction' ||
+            sendTestData[i].req.method === 'eth_sendTransaction')
+        ) {
+          continue
+        }
+        console.log('📍 Executing: ', sendTestData[i].req.method)
+        response = await axiosInstance.post('/', sendTestData[i].req)
+        if (response.data.error !== undefined && response.data.error !== null) {
+          console.error(
+            `❌ Invalid rpc call for : ${sendTestData[i].req.method} `,
+            response.data,
+            response.data.error
+          )
+          process.exit(1)
+        }
+
+        if (
+          sendTestData[i].req.method === 'eth_pendingTransactions' &&
+          response.data.result.length > 0 &&
+          !isValidTxHash(response.data.result)
+        ) {
+          console.error(
+            `❌ Invalid transaction hash returned : ${sendTestData[i].req.method} `,
```

Comment:
All the validation - per request - can be embedded into a single abstract method (something like `validateEth_SignTransaction`, or - generally speaking - `validate<requestMethod>`)